encoders/src/main/scala/au/csiro/pathling/encoders/SchemaConverter.scala

```diff
@@ -55,7 +56,14 @@ private[encoders] class SchemaConverterProcessor(override val fhirContext: FhirC
 
   override def buildComposite(definition: BaseRuntimeElementCompositeDefinition[_],
                               fields: Seq[StructField]): DataType = {
-    StructType(fields ++ createFidField() ++ createExtensionField(definition))
+    val updatedFields = definition.getImplementingClass match {
+      case cls if classOf[Quantity].isAssignableFrom(cls) => fields ++ Seq(
+        StructField("_value_canonicalized", DecimalCustomCoder.decimalType),
+        StructField("_code_canonicalized", DataTypes.StringType)
+      )
+      case _ => fields
+    }
+    StructType(updatedFields ++ createFidField() ++ createExtensionField(definition))
   }
 
```

Comment:
Short term maybe it would be better to refactor the quality fields creation (conditional) to a method similar to `createExtensionsFields` and write something lie:
`StructType(fields ++ createQuantityFields(definition) ++ createFidField() ++ createExtensionField(definition))`

Longer term this perhaps can be designed in such a way that all aspects of creation of synthetic (i.e. schema, ser, deser) is encapsulated in a single class, so that changes to that do not need to be tracked across three differenc classes (i.e. SchemaConverter, SerializderBuilder and DeserializerBuider).

Reply:
Refctored.

Long term solution captured in #842 [Review and harmonize encoding customisations]